# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/x64/Lowering-x64.cpp
# Commit: 444f4554ea94
# Full Hash: 444f4554ea9456909908f2872d1359495d9ff8d3
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-09-11 15:10:42
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 6.2: Establish dependency from tls for x86 callWithABI div/mod i64. r=lth
#   
#   x86 has few register so to do div/mod for i64 it call the runtime
#   and clobber almost all gp registers including WasmTlsReg.
#   To be able to call c++ runtime via Builtin thunk we need to set up WasmTlsReg.
# ==============================================================================

diff -r 5dd737d76ead -r 444f4554ea94 js/src/jit/x64/Lowering-x64.cpp
--- a/js/src/jit/x64/Lowering-x64.cpp	Fri Sep 11 09:14:07 2020 +0000
+++ b/js/src/jit/x64/Lowering-x64.cpp	Fri Sep 11 09:30:56 2020 +0000
@@ -352,6 +352,10 @@
   defineInt64Fixed(lir, div, LInt64Allocation(LAllocation(AnyRegister(rax))));
 }
 
+void LIRGeneratorX64::lowerWasmBuiltinDivI64(MWasmBuiltinDivI64* div) {
+  MOZ_CRASH("We don't use runtime div for this architecture");
+}
+
 void LIRGeneratorX64::lowerModI64(MMod* mod) {
   if (mod->isUnsigned()) {
     lowerUModI64(mod);
@@ -363,6 +367,10 @@
   defineInt64Fixed(lir, mod, LInt64Allocation(LAllocation(AnyRegister(rdx))));
 }
 
+void LIRGeneratorX64::lowerWasmBuiltinModI64(MWasmBuiltinModI64* mod) {
+  MOZ_CRASH("We don't use runtime mod for this architecture");
+}
+
 void LIRGeneratorX64::lowerUDivI64(MDiv* div) {
   LUDivOrModI64* lir = new (alloc()) LUDivOrModI64(
       useRegister(div->lhs()), useRegister(div->rhs()), tempFixed(rdx));